beacon-chain/rpc/lookup/blocker.go

```diff
@@ -70,81 +72,165 @@ type BeaconDbBlocker struct {
+			}
+			var root [32]byte
+			copy(root[:], headRoot)
+			return root, nil, nil
+		}
+		headRoot, err := blk.Block().HashTreeRoot()
+		if err != nil {
+			return [32]byte{}, nil, errors.Wrap(err, "could not get head block root")
 		}
+		return headRoot, blk, nil
 	case "finalized":
 		finalized := p.ChainInfoFetcher.FinalizedCheckpt()
+		if finalized == nil {
+			return [32]byte{}, nil, errors.New("received nil finalized checkpoint")
+		}
 		finalizedRoot := bytesutil.ToBytes32(finalized.Root)
 		blk, err = p.BeaconDB.Block(ctx, finalizedRoot)
 		if err != nil {
-			return nil, errors.New("could not get finalized block from db")
+			return [32]byte{}, nil, errors.Wrap(err, "could not retrieve finalized block")
 		}
-	case "genesis":
-		blk, err = p.BeaconDB.GenesisBlock(ctx)
+		return finalizedRoot, blk, nil
+	case "justified":
+		jcp := p.ChainInfoFetcher.CurrentJustifiedCheckpt()
+		if jcp == nil {
+			return [32]byte{}, nil, errors.New("received nil justified checkpoint")
+		}
+		justifiedRoot := bytesutil.ToBytes32(jcp.Root)
+		blk, err = p.BeaconDB.Block(ctx, justifiedRoot)
 		if err != nil {
-			return nil, errors.Wrap(err, "could not retrieve genesis block")
+			return [32]byte{}, nil, errors.Wrap(err, "could not retrieve justified block")
 		}
+		return justifiedRoot, blk, nil
 	default:
-		if bytesutil.IsHex(id) {
-			decoded, err := hexutil.Decode(string(id))
+		if bytesutil.IsHex([]byte(id)) {
+			var err error
+			rootSlice, err = bytesutil.DecodeHexWithLength(id, fieldparams.RootLength)
 			if err != nil {
 				e := NewBlockIdParseError(err)
-				return nil, &e
-			}
-			blk, err = p.BeaconDB.Block(ctx, bytesutil.ToBytes32(decoded))
-			if err != nil {
-				return nil, errors.Wrap(err, "could not retrieve block")
+				return [32]byte{}, nil, &e
 			}
 		} else if len(id) == 32 {
-			blk, err = p.BeaconDB.Block(ctx, bytesutil.ToBytes32(id))
-			if err != nil {
-				return nil, errors.Wrap(err, "could not retrieve block")
-			}
+			// Handle raw 32-byte root
+			rootSlice = []byte(id)
 		} else {
-			slot, err := strconv.ParseUint(string(id), 10, 64)
+			slot, err := strconv.ParseUint(id, 10, 64)
 			if err != nil {
 				e := NewBlockIdParseError(err)
-				return nil, &e
+				return [32]byte{}, nil, &e
 			}
+
+			// Get blocks by slot
 			blks, err := p.BeaconDB.BlocksBySlot(ctx, primitives.Slot(slot))
 			if err != nil {
-				return nil, errors.Wrapf(err, "could not retrieve blocks for slot %d", slot)
+				return [32]byte{}, nil, errors.Wrapf(err, "could not retrieve blocks for slot %d", slot)
 			}
-			_, roots, err := p.BeaconDB.BlockRootsBySlot(ctx, primitives.Slot(slot))
+			ok, roots, err := p.BeaconDB.BlockRootsBySlot(ctx, primitives.Slot(slot))
 			if err != nil {
-				return nil, errors.Wrapf(err, "could not retrieve block roots for slot %d", slot)
+				return [32]byte{}, nil, errors.Wrapf(err, "could not retrieve block roots for slot %d", slot)
+			}
+			if !ok || roots == nil || len(roots) == 0 {
```

Comment:
```suggestion
			if !ok || len(roots) == 0 {
```
`roots == nil` is redundant